pkg/controller/awsfederatedrole/awsfederatedrole_controller.go

```diff
@@ -115,6 +115,11 @@ func (r *ReconcileAWSFederatedRole) Reconcile(request reconcile.Request) (reconc
 	// Build custom policy in AWS-valid JSON and converts to string
 	jsonPolicy, err := utils.MarshalIAMPolicy(*instance)
 
+	// If AWSCustomPolicy and AWSManagedPolicies don't exist, exit reconcile
+	if len(instance.Spec.AWSManagedPolicies) == 0 && instance.Spec.AWSCustomPolicy.Name == "" {
+		return reconcile.Result{}, nil
```

Comment:
Should we error instead here or at least log a message? cc/ @fahlmant

Reply:
update condition and log err